src/components/GameItem.js

```diff
@@ -4,23 +4,45 @@ import ItemIcons from '../ItemIcons.js';
 import PropTypes from 'prop-types';
 
 class GameItem extends Component {
-  propTypes = {
+  constructor () {
+    super();
+    this.state = {
+      spottedType: '',
+    }
+  }
+
+  static propTypes = {
     height: PropTypes.number.isRequired,
     layer: PropTypes.number.isRequired,
   }
 
+  onTypeClick = () => {
+    let spottedType;
+    if (this.props.icon === 'litter') {
+      spottedType = 'spotted-litter';
```

Comment:
I'm not quite convinced by this callback's name. `onTypeClick` implies the user is clicking on a "type", which doesn't make sense. Maybe `onItemClick` or just `onClick` would be a stronger choice.